python/lsst/dax/imgserv/exceptions.py

```diff
@@ -0,0 +1,36 @@
+# This file is part of dax_imgserv.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+class UsageError(Exception):
+    def __init__(self, message=None):
```

Comment:
I'm a bit concerned that this is not calling the base constructor. Also, `self.message` doesn't seem to be something that is part of the normal Exception API. It's a Python 2-ism from what I can see.  I think you need to do something like:
```python
class ImageNotFound(Exception):
    def __init__(self, msg='Image Not Found'):
        super().__init__(msg)
```
This will then work like a normal Python 3 exception.

Reply:
I was following the official guide: https://docs.python.org/3.7/tutorial/errors.html

Reply:
Wow. Section 8.5 is weird because you end up with an exception that does not stringify like other exceptions and something where you have to know that the message is in `.message` so have to have special catch blocks for those exceptions (As you have). I don't know what to say.  The version I show above does stringify in the expected way...

Reply:
Please call it `ImageNotFoundError`

Reply:
Changed per your version - as it looks simpler.

Reply:
Seeing the other comment, I do wonder if defaulting the message is ever a good thing. Are the cases that trigger `ImageNotFoundError` always the same and there is no context information that is ever useful?

Reply:
I agree that no defaulting is better, as more contextual info can be provided.